go/contacts/resolution_notice.go

```diff
@@ -0,0 +1,69 @@
+package contacts
+
+import (
+	"context"
+
+	"github.com/keybase/client/go/encrypteddb"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type ContactResolution struct {
+	Description  string
+	ResolvedUser keybase1.User
+}
+
+func getKeyFn(mctx libkb.MetaContext) func(ctx context.Context) ([32]byte, error) {
+	keyFn := func(ctx context.Context) ([32]byte, error) {
+		return encrypteddb.GetSecretBoxKey(ctx, mctx.G(),
+			encrypteddb.DefaultSecretUI,
+			libkb.EncryptionReasonContactsResolvedServer,
+			"encrypting contact data for server")
+	}
+	return keyFn
+}
+
+func encryptContactBlob(mctx libkb.MetaContext, res ContactResolution) ([]byte,
+	error) {
+	return encrypteddb.EncodeBox(mctx.Ctx(), res, getKeyFn(mctx))
+}
+func DecryptContactBlob(mctx libkb.MetaContext,
+	contactResBlob []byte) (res ContactResolution, err error) {
+	err = encrypteddb.DecodeBox(mctx.Ctx(), contactResBlob,
+		getKeyFn(mctx), &res)
+	return res, err
+}
+
+// TODO: actually call this
+func SendContactResolutionToServer(mctx libkb.MetaContext,
+	resolutions []ContactResolution) error {
+
+	type resolvedArg struct {
+		ResolvedContactBlob []byte `json:"b"`
```

Comment:
base64 since we're sending json? and can check that its valid base64 in the web handler